gamemodes/terrortown/gamemode/shared/sh_player_ext.lua

```diff
@@ -146,6 +146,16 @@ function plymeta:SetRole(subrole, team, forceHooks, suppressEvent)
 		---
 		-- @realm shared
 		hook.Run("TTT2UpdateSubrole", self, oldSubrole, subrole)
+
+		if SERVER then
+			-- trigger the update in the next tick to make sure the role is updated
+			-- on the client first before any markerVision updates are sent
```

Comment:
Why is this important?
This just describes what the code already describes.

Reply:
Liie the first comment says: it makes sure that it is executed after the role updates is finished

Reply:
Yes but why?

Reply:
Because if the order is wrong, the client still has the old role information. I therefore either have to delay the transmission or send all the role data that might be needed along with this update

Reply:
But what makes it important that the clients role info is fully correct here? What calculations are done that are not reactive? Thats what the comment should describe

Reply:
Right now mainly the color. But it is up to the dev that puts their code in the hook.

Reply:
But the comment literally says that this is done to update the role on the client first. And this is the reason: update the role on the client first.

Adding a comment like "to make sure that the role is updated on the client first first that one can access the correct role" seems redundant

Reply:
You don't get what im saying here. Answering the why is important here. Explaining which cases otherwise fail is important. It is still not obvious why we would care about the order of updates. Also your example comment is badly worded and does not answer the question of why doing this.